### DISCUSSION TITLE
Elektrischer Energieverbrauch Lambda-WP per Modbus-Zähler auslesen und anzeigen

### DISCUSSION THREAD

**Discussion by SunnyTwinDad** [camueller/SmartApplianceEnabler#610] [Allgemein]
Hallo Community,
ich habe eine Lambda WP in Betrieb genommen und möchte nun im ersten Schritt den elektrischen Verbrauch dieser im SunnyPortal sehen können. Ein separater Modbus-Stromzähler ist nicht verbaut. Die Wärmepumpe zeigt aber auf dem Display der Reglereinheit einen entsprechenden Wert an. Dieser Wert wird bei mir leider noch dauerhaft mit "0" übertragen, auch wenn die Wärmepumpe läuft. Daher habe ich zu Testzwecken, einfach mal die "Vorlauftemperatur" als Leistungswert genutzt (Beispiel: 42,17 Grad -> 4217 Watt). Das funktioniert, wie der Screenshot belegen soll:

![WP_Verbrauch_Vorlauftemp](https://github.com/camueller/SmartApplianceEnabler/assets/170979314/8f73a470-ef9e-49f0-a28f-a6b3f63c5681)

Nun meine Frage: Die Werte für die "Wärmemenge Q_p" und der "COP" können ebenfalls per Register abgefragt werden. Die elektrische Leistung wäre ja Q_p / COP. Hier müsste man noch die Division durch 0 abfangen. Mir ist als Newbie noch nicht klar, wie man solche Vorabberechnungen mit SAE hinbekommt.

**Comment by camueller** [comment]
Der SAE unterstützt nicht die Abfrage mehrerer Register, um deren Werte matematisch zu verknüpfen. Entwerder Deine WP liefert die Leistung direkt als Wert oder Du baust einen separaten Zählen davor. Eine andere Lösung gibt es nicht.

**Comment by SunnyTwinDad** [comment]
Alles klar. 

Falls noch jemand auf das Thema stößt: Die WP muss zunächst per Register 102 (actual input power or actual excess power) mit 0 oder höher beschrieben werden (function code 16_dez = 0x10). Stand bei mir - aus welchen Gründen auch immer - auf "-32768"  [Watt].  Dann geht der Operating State (reg 101) auch aus dem Fehlerfall (Wert = 3) heraus. Und die Zählerabfrage per Register 103 funktioniert - dann ist auch keine mathematische Umrechnung erforderlich.
